### ISSUE TITLE
Plans to merge with the official CMake project?

### ISSUE THREAD

**Issue by adk9** [ellak-monades-aristeias/CMake-Docker#2]
Great work!

Are there plans to push this upstream to the CMake project?

Based on the questionnaire results, it seems like a large percentage of CMake and CPack users would like to see this feature merged in. Are there any pending issues holding the merge back?

**Comment by progtologist** [comment]
Yes there are a few issues holding it back, mostly related to the instability of docker itself (inconsistences in the documentation and a few bugs).
I am planning on getting this complete as soon as I find the time to get up to date with the latest docker advancements and have some spare time to implement those.
Given the current state of docker, I doubt this will be integrated in cmake by Kitware, I would have made the same decision myself, I wouldn't want something buggy to mess a really great and accepted software.

**Comment by do-m-en** [comment]
@progtologist Coincidentally enough I've also started looking into docker in combination with CMake 9 days ago :) and stumbled on your e-mail on CMake mailing list and added on my TODO list to ask if anybody knows what happened with this project. Imagine my surprise today when I stumbled across this repo :)
I'm new to docker and probably have some wrong impressions on how to use it but I'll try to use it on my hobby project.
I'll try to go over the code of your extensions over the next couple of days but since I'll be working on this on my free time I can't promise anything... Currently all the automated tests are failing so I'll try to get to the bottom of this first.
If the extensions will work for me maybe we could work together to get at least a part of it into upstream CMake - perhaps into 3.8 or 3.9 (of course depending on how much our time allows).
Have you already discussed it with guys from Kitware?

**Comment by progtologist** [comment]
@do-m-en nothing official has been discussed with Kitware as far as I remember, however I am also interested in giving this a lift to make it inside CMake.
Unfortunately docker has disappointed me over these months and has given me the impression that even if this makes it in CMake, no one can say for sure how long this will remain functioning, and how much effort it will take to maintain it.
I am also available only on my free time, as I am currently with my hands full with my occupation and my PhD, so I make no guarantees, but I would love to get back to this!
